mockagent_test.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (C) 2020 Storj Labs, Inc.
+// Copyright (C) 2014 Space Monkey, Inc.
+// See LICENSE for copying information.
+
+package jaeger
+
+import (
+	"context"
+	"net"
+	"sync"
+	"sync/atomic"
+
+	"github.com/apache/thrift/lib/go/thrift"
+
+	"storj.io/monkit-jaeger/gen-go/agent"
+	"storj.io/monkit-jaeger/gen-go/jaeger"
+)
+
+// StartMockAgent starts a mock agent on a local udp port.
+func StartMockAgent(hostPort string) (*MockAgent, error) {
+	addr, err := net.ResolveUDPAddr("udp", hostPort)
+	if err != nil {
+		return nil, err
+	}
+	conn, err := net.ListenUDP(addr.Network(), addr)
+	if err != nil {
+		return nil, err
+	}
+	var w sync.WaitGroup
+	mock := &MockAgent{
+		batches: make([]*jaeger.Batch, 0),
+		mux:     &sync.Mutex{},
+		conn:    conn,
+		addr:    conn.LocalAddr().String(),
+		started: &w,
+	}
+
+	w.Add(1)
+	go mock.serve()
```

Comment:
I think we need this to be:

```
go func(){
    mock.serve()
    w.Done()
}()
```

Reply:
I realized that we don't need this signal here